ng-dev/ts-circular-dependencies/index.ts

```diff
@@ -119,6 +107,33 @@ export function main(
     Log.warn(`   Please rerun with "--warnings" to inspect unresolved imports.`);
   }
 
+  if (goldenFile === undefined) {
+    if (approve) {
+      Log.error(
+        `x  Cannot approve circular depdencies within this repository as no golden file exists.`,
+      );
+      return 1;
+    }
+    if (cycles.length > 0) {
+      Log.error(`x  No circular dependencies are allow within this repository.`);
+      return 1;
+    }
+
+    Log.info(green('✔  No circular dependencies found in this repository.'));
+    return 0;
+  }
+
+  if (approve) {
+    writeFileSync(goldenFile, JSON.stringify(actual, null, 2));
+    Log.info(green('✔  Updated golden file.'));
+    return 0;
+  }
+
+  if (!existsSync(goldenFile)) {
```

Comment:
Wouldn't this always exist after writing before?

Reply:
The file would only be guaranteed to be written to, if an approval happened.  If its a regular check, then it wouldn't be written in the block above, and could technically not exist.  We verify it exists because just below here we read from it to do the actual comparison.